### PR TITLE
fix: Convert UUID DIDs to lower case UUID [DEV-1705] 

### PR DESCRIPTION
**Pull request by Toktar**

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-1705 Fix resource UUIDs (uppercase / lowercase error)](https://app.clickup.com/t/6600954/DEV-1705)

**Review by askolesov — COMMENTED** [review]
Really great work, thanks! There is just a couple of minor ideas I've just came up with. It can be implemented either by you or by me when tests are ready:
* I realized that we can keep keeper and handlers almost as is. To do it, we can normalize payload even before `.toDid` is called, in sth like `normalize msg create did` and so on.
* I suggest to move some functions to helpers. Like `normalize id` -> `id helper`, `normalize did` -> `did helper`

P.S. Separate comments above are for future me.
P.P.S. I don't approve only because of pending integration tests.

**Review by ankurdotb — COMMENTED** [review]
Left a comment on ClickUp on whether this invalidates signatures

File: x/cheqd/types/did_service.go [hunk 1]
```diff
@@ -63,3 +64,61 @@ func IsUniqueServiceListByIdRule() *CustomErrorRule {
 		return nil
 	})
 }
+
+func NormalizeIdentifier(did string) string {
```
[the remark this answers is not in the document]
Comment (Toktar) [reply x/cheqd/types/did_service.go]: Done

**Comment by Toktar** [comment]
@ankurdotb Do you have any questions about signatures for now?

**Review by askolesov — APPROVED** [review]
Cool, thanks!

**Pull request merged**